### PR TITLE
iotedge check improvements for nested edge

### PR DESCRIPTION
**Pull request by huguesBouvier**
In nested environment, the diagnostic image extracts the trust bundle before establishing the communication with upstream.
Improved error message with a suggestion for our user.

Before it was just trusting the upstream CA.

Removed unused publishing of connectivity test.
Test:
Put a dummy CA inside the trustbundle. Confirmed it fails 
Put good CA, confirmed it passes
Checked in regular, non nested scenario for regression.

### CODE REVIEW

**Review by daprilik — APPROVED** [review]
You might want to get another signoff from someone familiar with the C# bits, but at first glance this seems fine.

File: edge-util/src/Microsoft.Azure.Devices.Edge.Util/CertificateHelper.cs [hunk 1]
```diff
@@ -216,11 +216,10 @@ public static bool ValidateCertExpiry(X509Certificate2 certificate, ILogger logg
         public static void InstallCertificates(IEnumerable<X509Certificate2> certificateChain, ILogger logger)
         {
             X509Certificate2[] certs = Preconditions.CheckNotNull(certificateChain, nameof(certificateChain)).ToArray();
-            Preconditions.CheckNotNull(logger, nameof(logger));
 
             StoreName storeName = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? StoreName.CertificateAuthority : StoreName.Root;
 
-            logger.LogInformation($"Installing certificates {string.Join(",", certs.Select(c => $"[{c.Subject}:{c.GetExpirationDateString()}]"))} to {storeName}");
+            logger?.LogInformation($"Installing certificates {string.Join(",", certs.Select(c => $"[{c.Subject}:{c.GetExpirationDateString()}]"))} to {storeName}");
```
[the remark this answers is not in the document]
Comment (huguesBouvier) [reply edge-util/src/Microsoft.Azure.Devices.Edge.Util/CertificateHelper.cs:222]: The issue is that the diagnostic image doesn't have a logger. I want to reuse it, hence the change.

Additionnally, other functions in that same file are using the same null checking pattern. Now it is also consistent.

File: edge-modules/iotedge-diagnostics-dotnet/IotedgeDiagnosticsDotnet.sln [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+		{66C2EE0E-EF8D-4546-8B46-088C6981B7F1}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{66C2EE0E-EF8D-4546-8B46-088C6981B7F1}.Release|Any CPU.Build.0 = Release|Any CPU
+	EndGlobalSection
+	GlobalSection(SolutionProperties) = preSolution
+		HideSolutionNode = FALSE
+	EndGlobalSection
+	GlobalSection(ExtensibilityGlobals) = postSolution
+		SolutionGuid = {0515A383-449B-4D1B-862F-50A0A4F170B1}
+	EndGlobalSection
+EndGlobal
```
[the remark this answers is not in the document]
Comment (huguesBouvier) [reply edge-modules/iotedge-diagnostics-dotnet/IotedgeDiagnosticsDotnet.sln:36]: The diagnostic image is now dependent on the workload API to get the trust bundle. 
I create a solution that contains the 2 projects.

[the remark this answers is not in the document]
Comment (huguesBouvier) [reply edge-modules/iotedge-diagnostics-dotnet/IotedgeDiagnosticsDotnet.sln:36]: Thinking about it more, I think you are right, it is use only if you want to open them separately from the mega file containing all projects.
I'll remove it.

File: edge-modules/iotedge-diagnostics-dotnet/src/Program.cs [hunk 3]
```diff
@@ -71,33 +77,57 @@ static async Task EdgeAgent(string managementUri)
             }
         }
 
-        static async Task Upstream(string hostname, string port, string proxy, string isNested)
+        static async Task LoadTrustBundle(string workload_uri)
+        {
+            string dummy = "0";
```
[the remark this answers is not in the document]
Comment (huguesBouvier) [reply edge-modules/iotedge-diagnostics-dotnet/src/Program.cs:82]: It just creates a class that can be used to access several element from the API. We use only a subset from it.
Additionally the diagnostic image is not a module of iotedge, so it doesn't have module ID and such.

File: edge-modules/iotedge-diagnostics-dotnet/src/Program.cs [hunk 4]
```diff
@@ -71,33 +77,57 @@ static async Task EdgeAgent(string managementUri)
...
+
+            IEnumerable<X509Certificate2> trustBundle = await CertificateHelper.
+                GetTrustBundleFromEdgelet(new Uri(workload_uri), dummy, API_VERSION, dummy, dummy);
+            CertificateHelper.InstallCertificates(trustBundle, null);
+        }
+
+        static async Task Upstream(string hostname, string port, string proxy, string isNested, string workload_uri)
         {
+            bool nested = string.Equals(isNested, "true");
```
[the remark this answers is not in the document]
Comment (huguesBouvier) [reply edge-modules/iotedge-diagnostics-dotnet/src/Program.cs:91]: You won't, it is passed is has a parameter from the rust code. Rust knows nested is true if you have a parent defined

File: scripts/linux/buildBranch.sh [hunk 5]
```diff
@@ -22,7 +22,6 @@ SRC_E2E_TEST_FILES_DIR=$ROOT_FOLDER/e2e_test_files
 SRC_CERT_TOOLS_DIR=$ROOT_FOLDER/tools/CACertificates
 FUNCTIONS_SAMPLE_DIR=$ROOT_FOLDER/edge-modules/functions/samples
 VERSIONINFO_FILE_PATH=$BUILD_REPOSITORY_LOCALPATH/versionInfo.json
-CONNECTIVITY_TEST_SCRIPT_DIR=$ROOT_FOLDER/test/connectivity/scripts
```
[the remark this answers is not in the document]
Comment (huguesBouvier) [reply scripts/linux/buildBranch.sh:25]: yw :)

File: edge-modules/iotedge-diagnostics-dotnet/src/Program.cs [hunk 6]
```diff
@@ -2,17 +2,23 @@
...
     using System.Threading.Tasks;
     using Microsoft.Azure.Devices.Edge.Util;
     using Microsoft.Extensions.Configuration;
     using ProxyLib.Proxy;
 
     class Program
     {
+        const string API_VERSION = "2019-01-30";
+
         public static int Main(string[] args)
```
[the remark this answers is not in the document]
Comment (huguesBouvier) [reply edge-modules/iotedge-diagnostics-dotnet/src/Program.cs:22]: hum that I am not sure, it is standard starting point.
I tried, got error

File: edge-modules/iotedge-diagnostics-dotnet/src/Program.cs [hunk 7]
```diff
@@ -2,17 +2,23 @@
...
+    using System.Security.Authentication;
+    using System.Security.Cryptography.X509Certificates;
     using System.Threading.Tasks;
     using Microsoft.Azure.Devices.Edge.Util;
     using Microsoft.Extensions.Configuration;
     using ProxyLib.Proxy;
 
     class Program
     {
+        const string API_VERSION = "2019-01-30";
```
[the remark this answers is not in the document]
Comment (huguesBouvier) [reply edge-modules/iotedge-diagnostics-dotnet/src/Program.cs:20]: Fixed

File: edge-modules/iotedge-diagnostics-dotnet/src/Program.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (and-rewsmith) [reply edge-modules/iotedge-diagnostics-dotnet/src/Program.cs:82]: Oh I see you use this dummy value in multiple parameters of the GetTrustBundleFromEdgelet.

**Review by and-rewsmith — APPROVED** [review]

**Pull request merged**